src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.subsystems;
+
+import java.util.List;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatch;
+
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.I2C.Port;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ControlPanelSubsystem extends SubsystemBase {
+
+    public VictorSPX cpMotor;
+    public DoubleSolenoid cpSolenoid;
+    public ColorSensorV3 colorSensor;
+    public DriverStation ds;
+    public ColorMatch colorMatcher;
+
+    Color turnToColor = null;
+    public Color detectedColor = null;
+    public Color fmsColor = null;
+    Color lastColor = null;
+
+    List<String> colorNames;
+    public List<Color> colors;
+
+    //TODO: Both of these were "will_reset_to" variables in python, but I couldn't find a java alternative
+    public int speed = 0;
+    public DoubleSolenoid.Value solenoidState = DoubleSolenoid.Value.kReverse;
+
+    public ControlPanelSubsystem(){
+        cpMotor = new VictorSPX(2);
+        cpSolenoid = new DoubleSolenoid(5, 4);
```

Comment:
I wonder if the team should get in the habit of having a constants class that holds all the port numbers for the different electronics. It may be useful later on to ensure you don't change something in one class but forget to change it in another (e.g if the physical cables were swapped around)

Reply:
Maybe open an issue for it? That seems like something to focus a different pull request on.